### DISCUSSION TITLE
Unexpected behavior of "cache-only" vs "cache-first" request policy when using SSR

### DISCUSSION THREAD

**Discussion by ssimonitch** [urql-graphql/urql#2477] [Q&A] [answered]
Hello,

I'm not sure if this is a bug or a lack of understanding around how the different request policies work with urql's cache, but I have a NextJS project that uses `next-urql` for hydrating the urql cache during SSR and I have noticed that when I use the "cache-first" policy in my client-side fetching I immediately receive the cached data back whereas when I use "cache-only" I first receive `null` which I believe indicates nothing is found in the cache?

I would have expected "cache-only" to also immediately return a result instead of `null` similar to how "cache-first" is working. Am I missing something?

When I add logging to the urql client I also notice that with a "cache-first" request policy the first operation is of type `cacheMiss` but with the "cache-only" request policy the first operation is of type `cacheHit`.

Here's a code sandbox that's configured similarly to the project in question:
https://codesandbox.io/s/beautiful-allen-1rzcre?file=/pages/index.js

Thank you in advanced for the help!

**Comment by JoviDeCroock** [comment, marked as answer]
The thing is that `cache-only` will stop at the cache while on first-render your data resides in the `ssrExchange`, we could opt to stop at the `fetchExchange` however this can complicate things with other implementors like the `authExchange`/...
